transformer_engine/jax/flax/transformer.py

```diff
@@ -987,9 +987,8 @@ def __post_init__(self):
 
         if self.kernel_init is None:
             self.kernel_init = nn.initializers.variance_scaling(
-                1.0, "fan_in", "normal", self.weight_dtype
+                1.0, "fan_in", "normal", dtype=self.weight_dtype
             )
-            self.kernel_init = _kernel_init.astype(self.dtype)
```

Comment:
Looks like this was a bug from https://github.com/NVIDIA/TransformerEngine/pull/1469